### PR TITLE
reduce fixture-passing boilerplate in json-api tests

### PR DESCRIPTION
**Pull request by S11001001**
Successive extensions have left a plethora of `withHttpService` overloads with random pieces of data that have to be passed to utility functions. We should group this data together so utilities can be used without thinking about which random bits need to be passed along.

### CODE REVIEW

**Comment by S11001001** [comment]
@realvictorprm Please take a look at this draft; do you think we would be better off just defining newtypes and passing `url`, `encoder`, `decoder` _et al_ implicitly? I'm generally loath to use implicit params that aren't indexed by a type parameter, but the approach here is even less composable, and this is just test code. Though "this is just test code" also speaks to not worrying too much about composability.

Either way would serve the goal of "don't think about the data you got from `withHttpService` too hard; just focus on writing your test and not what random bits you need to pass around"; please evaluate both possibilities with that in mind.

**Review by realvictorprm — COMMENTED** [review]
> @realvictorprm Please take a look at this draft; do you think we would be better off just defining newtypes and passing url, encoder, decoder et al implicitly? I'm generally loath to use implicit params that aren't indexed by a type parameter, but the approach here is even less composable, and this is just test code. Though "this is just test code" also speaks to not worrying too much about composability.

> Either way would serve the goal of "don't think about the data you got from withHttpService too hard; just focus on writing your test and not what random bits you need to pass around"; please evaluate both possibilities with that in mind.

I think passing in just a context object with making use of imports is a better way than before to handle the situation. Regarding implicits I can tell, that I'm currently in favor of using slightly less implicits to increase code clarity.
Furthermore I went through the PR changes thoroughly and if I understood the changes correctly then the user token tests currently are probably not passing completely anymore (due to the call to `unqiuePartyWithAuthHeader` being fixed to the implementation for non user/custom tokens).
Anyways, need to stress again that this definitely _is_ an improvement.

**Comment by realvictorprm** [comment]
Important thing I forgot to add: Thanks for trying to improve this @S11001001, very much appreciated! :)

**Comment by S11001001** [comment]
> Regarding implicits I can tell, that I'm currently in favor of using slightly less implicits to increase code clarity.

If this was main code I would probably be trying a newtype+implicits approach. But this is test code so I'm experimenting a little. The traits truly are a mess that make them wrong for a main code design, but I'm happy with them when we're just throwing things together here.

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 1]
```diff
@@ -410,37 +414,34 @@ abstract class AbstractHttpServiceIntegrationTestTokenIndependent
...
-            uri,
-            encoder,
-            headers,
-          ).map(inside(_) { case Seq(domain.ActiveContract(_, _, _, JsObject(fields), _, _, _)) =>
-            fields.get("currency") should ===(Some(JsString(testCurrency)))
-          })
-        }
+    s"query record contains handles '$testLbl' strings properly" in withHttpService { fixture =>
+      fixture.getUniquePartyAndAuthHeaders("Alice").flatMap { case (alice, headers) =>
+        searchExpectOk(
```
[the remark this answers is not in the document]
Comment (realvictorprm) [reply ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:419]: > By the way I forgot to suggest to check "ignore whitespace" upside_down_face

I should have done that, however I do admit that I noticed way too late that this cannot be merged yet :sweat_smile: So I was in my usual mode of also checking whitespace if I run into them.

**Pull request merged**